### ISSUE TITLE
some questions about scatter 

### ISSUE THREAD

**Issue by YOLOsyb** [pytorch/pytorch#37130]
`import torch

### cpu
input = torch.zeros(8, 3, 7, 10, dtype=torch.float)
src = torch.randn(8, 3, 7, 10, dtype=torch.float)
index = torch.randint(0, 3, [8, 3 ,7, 10])
dim = 1
input.scatter_(dim, index, src)

### gpu
ginput = torch.zeros(8, 3, 7, 10, dtype=torch.float).cuda()
gsrc = src.cuda()
gindex = index.cuda()
ginput.scatter_(dim, gindex, gsrc)

err = (ginput.cpu() - input).sum()
print(err)`

### result
tensor(3.76)

### 
My question is why the result is inconsistent?

**Comment by ngimel** [comment]
If there are repeated indices, `scatter` results are not deterministic. Setting docs tag to document this.

**Comment by YOLOsyb** [comment]
@ngimel  Thanks for your answer!

I'm still a little confused about the code. 

https://github.com/pytorch/pytorch/blob/b889e0da8ab33e0b47abc817e679851299728be7/aten/src/THC/THCTensorScatterGather.cu#L113

`  for (IndexType linearId = blockIdx.x * blockDim.x + threadIdx.x;
       linearId < totalElements;
       linearId += gridDim.x * blockDim.x) {`

I suspect gpu parallelism。
